lustre_exporter_test.go

```diff
@@ -0,0 +1,1362 @@
+// (C) Copyright 2017 Hewlett Packard Enterprise Development LP
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/HewlettPackard/lustre_exporter/sources"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/expfmt"
+)
+
+type labelPair struct {
+	Name  string
+	Value string
+}
+
+type promType struct {
+	Name   string
+	Help   string
+	Type   int
+	Labels []labelPair
+	Value  float64
+	Parsed bool // Parsed returns 'true' if the metric has already been parsed
+}
+
+const (
+	// Constants taken from https://github.com/prometheus/client_model/blob/master/go/metrics.pb.go
+	counter = 0
+	gauge   = 1
+	untyped = 3
+)
```

Comment:
Could we import these instead of redefining them? Probably not without changing the source to allow for them to be external, but worth considering in the future, maybe. Though as I type this, I'm realizing how much work it would be for very little gain.